src/test/java/teammates/test/pageobjects/InstructorCourseStudentDetailsEditPage.java

```diff
@@ -73,10 +73,19 @@ private void fillStudentDetailsForm(String studentName, String teamName, String
             fillTextBox(commentsTextbox, comments);
         }
         if (teamName == null) {
-            click(submitButton);
+            if (studentEmail == null) {
+                click(submitButton);
+            } else {
+                clickAndConfirm(submitButton);
+            }
+            
         } else {
             // if team name is edited, the confirmation dialog will pop up
-            clickAndConfirm(submitButton);
+            if (studentEmail == null) {
+                clickAndConfirm(submitButton);
+            } else {
+                clickAndConfirmTwoTimes(submitButton);
```

Comment:
I see what you mean. Then why don't we implement a variant of `clickAndConfirm` that does _not_ wait for that disappearance?
```java
clickAndConfirmWithoutWaitingForModalDisappearance(submitButton); // implement accordingly
if (studentEmail != null) {
    clickNoOnModal(); // implement accordingly
}
```